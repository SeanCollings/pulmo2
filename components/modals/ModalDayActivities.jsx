import React from 'react';
import {
  StyleSheet,
  View,
  Text,
  ScrollView,
  TouchableOpacity,
} from 'react-native';
import MaterialCommunityIcons from 'react-native-vector-icons/MaterialCommunityIcons';

import CustomModal from '.';
import { useTheme } from '../../hooks/useTheme';
import { getColour, getIcon } from '../../utils';

const ModalDayActivities = ({
  cancelModel,
  header,
  activities,
  navigation,
}) => {
  const theme = useTheme();

  const activitySelectedHandler = ({ date, favourite, title }) => {
    cancelModel();
    navigation.navigate('Activity', {
      item: { date, favourite, title },
    });
  };

  const opacity = theme.DARK ? 0.87 : 1;

  return (
    <CustomModal
      cancelModel={cancelModel}
      header={header}
      cancelTitle="Back"
      headingColour={theme.SECONDARY}
      animationType="none"
    >
      <View style={styles.container}>
        <ScrollView>
          {activities.map((activity) => {
            return (
              <TouchableOpacity
                key={activity.date}
                style={styles.contentContainer}
                onPress={() => activitySelectedHandler(activity)}
              >
                <View>
                  <View style={styles.rowContainer}>
                    <View style={styles.iconContainer}>
                      <MaterialCommunityIcons
                        name={getIcon(activity.type)}
                        color={getColour(activity.type, theme)}
                        size={20}
                      />
                    </View>
                    <Text
                      style={{
                        ...styles.contentText,
                        color: theme.TEXT,
                        opacity,
                      }}
                    >
                      {activity.title}
                    </Text>
                    <View style={{ ...styles.iconContainer, opacity }}>
                      <MaterialCommunityIcons
                        name="chevron-right"
                        color={theme.DARK ? theme.TEXT : theme.BORDER}
                        size={25}
                      />
                    </View>
                  </View>
                </View>
              </TouchableOpacity>
            );
          })}
        </ScrollView>
      </View>
    </CustomModal>
  );
};

const styles = StyleSheet.create({
  container: {
    width: '100%',
    maxHeight: 260,
  },
  contentContainer: {
    paddingHorizontal: 10,
    paddingTop: 10,
  },
  rowContainer: {
    flexDirection: 'row',
    justifyContent: 'space-between',
  },
  iconContainer: {
    justifyContent: 'center',
    width: '10%',
  },
  contentText: {
    paddingLeft: 10,
    textAlign: 'left',
    fontSize: 16,
    fontFamily: 'tit-light',
    width: '80%',
  },
});

export default ModalDayActivities;
