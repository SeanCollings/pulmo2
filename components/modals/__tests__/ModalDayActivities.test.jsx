import React from 'react';
import { act, create } from 'react-test-renderer';
import { cleanup, fireEvent } from 'react-native-testing-library';
import ModalDayActivities from '../ModalDayActivities';

jest.useFakeTimers();

describe('ModalDayActivities - unit test', () => {
  let component;
  const cancelModal = jest.fn();
  const navigation = { navigate: jest.fn() };
  const activities = [
    {
      date: '2020-07-26T15:15:43.411Z',
      type: 'Strength',
      title: 'Excercise Title',
      favourite: false,
    },
    {
      date: '2020-07-26T16:16:43.411Z',
      type: 'Strength',
      title: 'Excercise Title 2',
      favourite: true,
    },
  ];
  const header = 'ModalDayActivities';

  afterEach(() => {
    cleanup();
  });

  test('should render with custom props', async () => {
    await act(async () => {
      component = create(
        <ModalDayActivities
          cancelModal={cancelModal}
          header={header}
          navigation={navigation}
          activities={activities}
        />
      );
    });

    expect(component).toMatchSnapshot();
  });

  test('should allow the selection of an activity', async () => {
    await act(async () => {
      component = create(
        <ModalDayActivities
          cancelModal={cancelModal}
          header={header}
          navigation={navigation}
          activities={activities}
        />
      );
    });

    const activitySelectButton = component.root.find(
      ({ props }) => props.testID === `activity_select_${activities[0].date}`
    );

    fireEvent.press(activitySelectButton);
    expect(navigation.navigate).toHaveBeenCalled();
    expect(cancelModal).toHaveBeenCalled();
  });
});
