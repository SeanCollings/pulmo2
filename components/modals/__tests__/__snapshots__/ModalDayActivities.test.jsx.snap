// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalDayActivities - unit test should render with custom props 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "rgba(0, 0, 0, 0.5)",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Modal
    animationType="none"
    hardwareAccelerated={false}
    transparent={true}
    visible={true}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "rgba(0, 0, 0, 0.5)",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "white",
            "borderRadius": 12,
            "elevation": 5,
            "padding": 20,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.25,
            "shadowRadius": 2.84,
            "width": "80%",
          }
        }
      >
        <View
          style={
            Object {
              "flexDirection": "row",
              "justifyContent": "space-around",
              "width": "105%",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#002f56",
                "fontFamily": "tit-regular",
                "fontSize": 20,
                "textAlign": "center",
                "width": "80%",
              }
            }
          >
            ModalDayActivities
          </Text>
        </View>
        <View
          style={
            Object {
              "width": "100%",
            }
          }
        >
          <View
            style={
              Object {
                "maxHeight": 260,
                "width": "100%",
              }
            }
          >
            <RCTScrollView>
              <View>
                <View
                  accessible={true}
                  focusable={true}
                  isTVSelectable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "opacity": 1,
                      "paddingHorizontal": 10,
                      "paddingTop": 10,
                    }
                  }
                  testID="activity_select_2020-07-26T15:15:43.411Z"
                >
                  <View>
                    <View
                      style={
                        Object {
                          "flexDirection": "row",
                          "justifyContent": "space-between",
                        }
                      }
                    >
                      <View
                        style={
                          Object {
                            "justifyContent": "center",
                            "width": "10%",
                          }
                        }
                      >
                        <Text
                          allowFontScaling={false}
                          style={
                            Array [
                              Object {
                                "color": "#00b8bb",
                                "fontSize": 20,
                              },
                              undefined,
                              Object {
                                "fontFamily": "material-community",
                                "fontStyle": "normal",
                                "fontWeight": "normal",
                              },
                              Object {},
                            ]
                          }
                        >
                          
                        </Text>
                      </View>
                      <Text
                        style={
                          Object {
                            "color": "#3e4b57",
                            "fontFamily": "tit-light",
                            "fontSize": 17,
                            "opacity": 1,
                            "paddingLeft": 10,
                            "textAlign": "left",
                            "width": "80%",
                          }
                        }
                      >
                        Excercise Title
                      </Text>
                      <View
                        style={
                          Object {
                            "justifyContent": "center",
                            "opacity": 1,
                            "width": "10%",
                          }
                        }
                      >
                        <Text
                          allowFontScaling={false}
                          style={
                            Array [
                              Object {
                                "color": "#CCC",
                                "fontSize": 25,
                              },
                              undefined,
                              Object {
                                "fontFamily": "material-community",
                                "fontStyle": "normal",
                                "fontWeight": "normal",
                              },
                              Object {},
                            ]
                          }
                        >
                          
                        </Text>
                      </View>
                    </View>
                  </View>
                </View>
                <View
                  accessible={true}
                  focusable={true}
                  isTVSelectable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "opacity": 1,
                      "paddingHorizontal": 10,
                      "paddingTop": 10,
                    }
                  }
                  testID="activity_select_2020-07-26T16:16:43.411Z"
                >
                  <View>
                    <View
                      style={
                        Object {
                          "flexDirection": "row",
                          "justifyContent": "space-between",
                        }
                      }
                    >
                      <View
                        style={
                          Object {
                            "justifyContent": "center",
                            "width": "10%",
                          }
                        }
                      >
                        <Text
                          allowFontScaling={false}
                          style={
                            Array [
                              Object {
                                "color": "#00b8bb",
                                "fontSize": 20,
                              },
                              undefined,
                              Object {
                                "fontFamily": "material-community",
                                "fontStyle": "normal",
                                "fontWeight": "normal",
                              },
                              Object {},
                            ]
                          }
                        >
                          
                        </Text>
                      </View>
                      <Text
                        style={
                          Object {
                            "color": "#3e4b57",
                            "fontFamily": "tit-light",
                            "fontSize": 17,
                            "opacity": 1,
                            "paddingLeft": 10,
                            "textAlign": "left",
                            "width": "80%",
                          }
                        }
                      >
                        Excercise Title 2
                      </Text>
                      <View
                        style={
                          Object {
                            "justifyContent": "center",
                            "opacity": 1,
                            "width": "10%",
                          }
                        }
                      >
                        <Text
                          allowFontScaling={false}
                          style={
                            Array [
                              Object {
                                "color": "#CCC",
                                "fontSize": 25,
                              },
                              undefined,
                              Object {
                                "fontFamily": "material-community",
                                "fontStyle": "normal",
                                "fontWeight": "normal",
                              },
                              Object {},
                            ]
                          }
                        >
                          
                        </Text>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </RCTScrollView>
          </View>
        </View>
        <View
          style={
            Object {
              "flexDirection": "row",
              "justifyContent": "space-around",
              "marginTop": 20,
              "width": "100%",
            }
          }
        >
          <View
            style={
              Object {
                "width": "40%",
              }
            }
          >
            <View
              accessible={true}
              focusable={true}
              isTVSelectable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Object {
                    "backgroundColor": "white",
                    "borderColor": "#CCC",
                    "borderRadius": 8,
                    "borderWidth": 1,
                    "justifyContent": "center",
                    "paddingHorizontal": 14,
                    "paddingVertical": 8,
                    "shadowColor": "#000",
                    "shadowOffset": Object {
                      "height": 2,
                      "width": 0,
                    },
                    "shadowOpacity": 0.25,
                    "shadowRadius": 2.84,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "#3e4b57",
                      "fontFamily": "tit-regular",
                      "fontSize": 14,
                      "opacity": 1,
                      "textAlign": "center",
                      "textTransform": "uppercase",
                    }
                  }
                >
                  Back
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </Modal>
</View>
`;
